numpy/_core/src/multiarray/dtypemeta.c

```diff
@@ -1230,6 +1233,11 @@ dtypemeta_wrap_legacy_descriptor(
         dtype_class->flags |= NPY_DT_NUMERIC;
     }
 
+    /* If sorting compare not defined, set to arrfunc default */
+    if (dt_slots->sort_compare == NULL) {
```

Comment:
maybe we should object and error out if this is a non-legacy dtype?

Reply:
Good idea, thanks! Did this.